### PR TITLE
Devel/analysis v3

### PR DESCRIPTION
**Pull request by stephlazar**
Major changes to the analysis_v3 framework. 

Changes are only related to analysis_v3 files, and to RB measurements which use this analysis.
(There is one change in UHFQC_multi_detector where correlated is now default False)

I added the demo notebook I presented in April during the software meeting on analysis_v3.
This is in PycQED_py3/examples/new_features_demos

### CODE REVIEW

File: pycqed/analysis_v3/processing_pipeline.py [hunk 1]
```diff
@@ -1,353 +1,774 @@
...
+              'keys_in': 'previous qb2.average_data1',
+              'std_keys': 'previous qb2.get_std_deviation1',
+              'node_name': 'SingleQubitRBAnalysis'}]
+    
+    Creating the pipeline:
+        - the analysis framework always expects keys_in to be a list of 
+         keys in the data_dict, and most functions expect keys_out
+        - to create the pipeline that will be used by the analysis 
+         framework, the user must call: 
+         ProcessingPipeline_instance(meas_obj_value_names_map), where
```
Comment (chellings) [inline pycqed/analysis_v3/processing_pipeline.py:93]: this documentation might be outdated (seems that meas_obj_value_names_map is no longer passed at init stage)

**Review by chellings — APPROVED** [review]
I have not checked the code in detail, but:
- we have seen the demo of analysis_v3, and it makes sense to make this part of the master to encourage people to start getting experience with it
- the RB code has been used successfully on XLD by @nathlacroix and it makes sense to have this in the master (which requires to merge also the analysis_v3 changes)
- the changes do not seem to have an influence on any other current measurements, so it should be save to merge

So, thank you for creating this pull request and for all your work on v3. Only one comment on documentation.

File: pycqed/analysis_v3/processing_pipeline.py (hunk 1 above)
Comment (stephlazar) [reply pycqed/analysis_v3/processing_pipeline.py:93]: It is not completely outdated. The pipeline instance must still be called with meas_obj_value_names map in order to resolve it. However, now I added the option to pass the un-resolved pipeline to the analysis and the analysis resolves it based on meas_obj_value_names_map. So the user does not necessarily need to do it, and I've updated this part of the docstring to say so.

**Comment by stephlazar** [comment]
Thanks a lot for looking at it!

**Comment by stephlazar** [comment]
I intentionally did not delete the branch.

**Pull request merged**